### ISSUE TITLE
ludwig serve Internal error 500 (model has no max_length, division by zero)

### ISSUE THREAD

**Issue by pkpro** [ludwig-ai/ludwig#3594]
**Describe the bug**
I tried to do the fine-tuning on CodeLlama-13b-Python-hf (converted from CodeLlama-13b-Python) with ludwig (4bit QLoRA), following your training video. The training was Ok. However I'm unable to serve the model from results/<experiment>/model because of the following error on any request:

```
Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.
~/.local/lib/python3.10/site-packages/ludwig/features/feature_utils.py:102: RuntimeWarning: divide by zero encountered in log
  return np.sum(np.log(sequence_probabilities))
Failed to run predict: Out of range float values are not JSON compliant
Traceback (most recent call last):
  File "~/.local/lib/python3.10/site-packages/ludwig/serve.py", line 92, in predict
    return NumpyJSONResponse(resp)
  File "~/.local/lib/python3.10/site-packages/starlette/responses.py", line 196, in __init__
    super().__init__(content, status_code, headers, media_type, background)
  File "~/.local/lib/python3.10/site-packages/starlette/responses.py", line 55, in __init__
    self.body = self.render(content)
  File "~/.local/lib/python3.10/site-packages/ludwig/utils/server_utils.py", line 168, in render
    return json.dumps(
  File "/usr/lib/python3.10/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib/python3.10/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.10/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
ValueError: Out of range float values are not JSON compliant
INFO:     <IP>:43490 - "POST /predict HTTP/1.1" 500 Internal Server Error
```

However in description.json and model_hyperparameters.json max_length is set to 32.

**To Reproduce**

1. Convert CodeLlama-13b-Python to hf
2. tune a little to have the result/<experiment>.
1. Try to serve the result
1. Send a request to port 8000

IDK if following is anyhow related:

Training was done with temperature 0.0. 

```yaml
model_type: llm
base_model: ./CodeLlama-13b-Python-hf

input_features:
  - name: instruction
    type: text

output_features:
  - name: output
    type: text

prompt:
  template: >-
    Below is an instruction that describes a task, paired with an input
    that may provide further context. Write a response that appropriately
    completes the request.

    ### Instruction: {instruction}

    ### Input: {input}

    ### Response:

generation:
  max_new_tokens: 4096
  temperature: 0.0 # Temperature is used to control the randomness of predictions.
  do_sample: True

preprocessing:
  split:
    type: fixed

adapter:
  type: lora

quantization:
  bits: 4

trainer:
  type: finetune
  epochs: 1
  batch_size: 1
  eval_batch_size: 2
  gradient_accumulation_steps: 16
  learning_rate: 0.00001
  optimizer:
    type: adamw
    params:
      eps: 1.e-8
      betas:
        - 0.9
        - 0.999
      weight_decay: 0
  learning_rate_scheduler:
    warmup_fraction: 0.03
    reduce_on_plateau: 0
```

Training is set to 0.1 in the json config files for inference.

**Expected behavior**
200 return code on request instead of 500

**Environment (please complete the following information):**

- OS: linux (gentoo and ubuntu 22.04)
- Version: seem to be irrelevant
- Python version: 3.11 and 3.10
- Ludwig version: 8.2

**Comment by jeffkinnison** [comment]
Hi @pkpro! I'll be looking into this today. I just have one followup for now: Can you provide the specific code/commands used in the reproduction steps? That would greatly help in tracking this down. Thank you!

**Comment by pkpro** [comment]
Hi @jeffkinnison , sorry for being so imprecise in the description. I used ludwig serve and ran a curl against it. I suppose the probelm is, that I have set a padding_token in the python script, which should be the same as eos_token for LLaMA2 models, and this is what caused the problem. I already removed the finte-tuned result so I cant really reproduce it myself ether. I will close this ticket and I'll fine-tune a smaller model next Saturday. And if the problem repeats I'll provide more detailed information about the fine-tuning and how and what do I ran along with the fine-tuned model.
